src/pybind/mgr/volumes/fs/operations/versions/subvolume_v1.py

```diff
@@ -54,6 +54,19 @@ def path(self):
     def features(self):
         return [SubvolumeFeatures.FEATURE_SNAPSHOT_CLONE.value, SubvolumeFeatures.FEATURE_SNAPSHOT_AUTOPROTECT.value]
 
+    def mark_subvolume(self):
+        # set subvolume attr, on subvolume root, marking it as a CephFS subvolume
+        # subvolume root is where snapshots would be taken, and hence is the <uuid> dir for v1 subvolumes
+        xattr_val = 1
+        try:
+            self.fs.setxattr(self.path, 'ceph.dir.subvolume', str(xattr_val).encode('utf-8'), os.XATTR_CREATE)
```

Comment:
I think XATTR_{CREATE,REPLACE} checks are not available for ceph vxatts -- this would never return EEXIST.
 
So, either we just not bother using this flag or check if the subvolume is already marked in Server::handle_set_vxattr() when handling `ceph.dir.subvolume`:

@ukernel @batrick

Reply:
I think removing the flag makes sense. Logically all directories have this "flag" either off or on. So this xattr cannot be created.

Reply:
this gets called when accessing a subvolume. that would lead to disk IO each time -- that's the reason I proposed to include this flag.

Reply:
In the `open` method? Perhaps we should just move the `mark_subvolume` call to quota/snapshot creation?

Reply:
I think it is ok (for now) to leave it here: handle_set_vxattr() for ceph.dir.subvolume safely handles subvolume mark on already marked subvolume: https://github.com/ceph/ceph/blob/master/src/mds/Server.cc#L5633

So it doesn't hit the disk every time -- just that it won't return EEXIST.